### PR TITLE
Tour content

### LINKED ISSUE #2199: Read tour content from .yaml
Read content for tour from yaml file instead of coding it in JSON in the template

### LINKED ISSUE #2200: Build tour for listing page
- [ ] create content for tour on snap listing page
- [ ] add button to trigger the tour

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #2199 
Fixes #2200 

Adds full content of publisher snap listing page tour.
Moves tour content from template to YAML file.
Makes it possible to target multiple elements with one tour step for better control over what is highlighted on the page.

### QA
- ./run or https://snapcraft-io-canonical-web-and-design-pr-2213.run.demo.haus/
- sign in and go to publisher listing page of any snap
- click on a tour icon (?) in right bottom corner
- go through the tour
- make sure it works fine when scrolling, resizing, etc
- try different snaps (with different amount of data filled in already)

Screenshot 2019-08-27 at 09 19 23

### CODE REVIEW

**Comment by bartaz** [comment]
@hughesjordan 

For the final step of the tour we need proper links for blog post and contact. If we don't have those we should probably rephrase this step to remove them.

Screenshot 2019-08-27 at 09 30 18

File: static/js/publisher/tour/tourOverlay.js [hunk 1]
```diff
@@ -71,13 +94,14 @@ export default function TourOverlay({ steps, onHideClick }) {
   // when current step changes, scroll step element into view
   useEffect(
     () => {
-      step.el.scrollIntoView({ block: "center", behavior: "smooth" });
+      // TODO: consider scrolling based on mask position instead of element
+      step.elements[0].scrollIntoView({ block: "center", behavior: "smooth" });
```
Comment (Caleb-Ellis) [inline static/js/publisher/tour/tourOverlay.js]: This causes highlighted content to appear below the viewport if it's taller than half the viewport height.

File: static/js/publisher/tour/tourOverlay.js [hunk 2]
```diff
@@ -48,18 +48,41 @@ const getMaskFromRect = rect => {
   };
 };
 
-// find DOM element for each step, ignore steps with no element
+// get mask that is an union of all elements' masks
+// calculates the rectangle that contains each individual element rectangles
+const getMaskFromElements = elements => {
+  return elements.map(el => getMaskFromRect(getRectFromEl(el))).reduce(
```
Comment (Caleb-Ellis) [inline static/js/publisher/tour/tourOverlay.js]: This is a little bit hard to parse - map into two nested functions into reduce. I guess it's not so hard to understand what goes into this function and what comes out, but the internals are a bit hazy for me.

Comment (bartaz) [reply static/js/publisher/tour/tourOverlay.js]: Wanted to format it nicer for better readability, but prettier wants it all in one line ;)
